# Assignment 07

In [1]:
import pandas as pd
import seaborn as sns


Matplotlib is building the font cache; this may take a moment.


In [2]:
data_house = pd.read_csv("./general_data.csv")
data_house.head()

Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   
4   32        No      Travel_Rarely  Research & Development                10   

   Education EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0          2  Life Sciences              1           1  Female  ...   
1          1  Life Sciences              1           2  Female  ...   
2          4          Other              1           3    Male  ...   
3          5  Life Sciences              1           4    Male  ...   
4          1        Medical              1           5    Male  ...   

   NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0                 1.0      Y                11              8   
1                 0.0      Y                23              8   
2                 1.0      Y                15              8   
3                 3.0      Y                11              8   
4                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

## Treating the data

In [11]:
data_house.isnull()

Age  Attrition  BusinessTravel  Department  DistanceFromHome  \
0     False      False           False       False             False   
1     False      False           False       False             False   
2     False      False           False       False             False   
3     False      False           False       False             False   
4     False      False           False       False             False   
...     ...        ...             ...         ...               ...   
4405  False      False           False       False             False   
4406  False      False           False       False             False   
4407  False      False           False       False             False   
4408  False      False           False       False             False   
4409  False      False           False       False             False   

      Education  EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0         False           False          False       False   False  ...   
1         False           False          False       False   False  ...   
2         False           False          False       False   False  ...   
3         False           False          False       False   False  ...   
4         False           False          False       False   False  ...   
...         ...             ...            ...         ...     ...  ...   
4405      False           False          False       False   False  ...   
4406      False           False          False       False   False  ...   
4407      False           False          False       False   False  ...   
4408      False           False          False       False   False  ...   
4409      False           False          False       False   False  ...   

      NumCompaniesWorked  Over18  PercentSalaryHike  StandardHours  \
0                  False   False              False          False   
1                  False   False              False          False   
2                  False   False              False          False   
3                  False   False              False          False   
4                  False   False              False          False   
...                  ...     ...                ...            ...   
4405               False   False              False          False   
4406               False   False              False          False   
4407               False   False              False          False   
4408               False   False              False          False   
4409               False   False              False          False   

      StockOptionLevel  TotalWorkingYears  TrainingTimesLastYear  \
0                False              False                  False   
1                False              False                  False   
2                False              False                  False   
3                False              False                  False   
4                False              False                  False   
...                ...                ...                    ...   
4405             False              False                  False   
4406             False              False                  False   
4407             False              False                  False   
4408             False              False                  False   
4409             False               True                  False   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0              False                    False                 False  
1              False                    False                 False  
2              False                    False                 False  
3              False                    False                 False  
4              False                    False                 False  
...              ...                      ...                   ...  
4405           False                    False                 False  
4406           False    

In [12]:
data_house.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
4405    False
4406    False
4407    False
4408    False
4409    False
Length: 4410, dtype: bool

### Since data has no duplicate values no need to drop duplicate values

## Analysing the data

In [29]:
print("The given dataset has following columns:")
for heads in data_house.columns:
    print("-", heads)

The given dataset has following columns:
- Age
- Attrition
- BusinessTravel
- Department
- DistanceFromHome
- Education
- EducationField
- EmployeeCount
- EmployeeID
- Gender
- JobLevel
- JobRole
- MaritalStatus
- MonthlyIncome
- NumCompaniesWorked
- Over18
- PercentSalaryHike
- StandardHours
- StockOptionLevel
- TotalWorkingYears
- TrainingTimesLastYear
- YearsAtCompany
- YearsSinceLastPromotion
- YearsWithCurrManager


In [14]:
data_house.describe()

Age  DistanceFromHome    Education  EmployeeCount   EmployeeID  \
count  4410.000000       4410.000000  4410.000000         4410.0  4410.000000   
mean     36.923810          9.192517     2.912925            1.0  2205.500000   
std       9.133301          8.105026     1.023933            0.0  1273.201673   
min      18.000000          1.000000     1.000000            1.0     1.000000   
25%      30.000000          2.000000     2.000000            1.0  1103.250000   
50%      36.000000          7.000000     3.000000            1.0  2205.500000   
75%      43.000000         14.000000     4.000000            1.0  3307.750000   
max      60.000000         29.000000     5.000000            1.0  4410.000000   

          JobLevel  MonthlyIncome  NumCompaniesWorked  PercentSalaryHike  \
count  4410.000000    4410.000000         4391.000000        4410.000000   
mean      2.063946   65029.312925            2.694830          15.209524   
std       1.106689   47068.888559            2.498887           3.659108   
min       1.000000   10090.000000            0.000000          11.000000   
25%       1.000000   29110.000000            1.000000          12.000000   
50%       2.000000   49190.000000            2.000000          14.000000   
75%       3.000000   83800.000000            4.000000          18.000000   
max       5.000000  199990.000000            9.000000          25.000000   

       StandardHours  StockOptionLevel  TotalWorkingYears  \
count         4410.0       4410.000000        4401.000000   
mean             8.0          0.793878          11.279936   
std              0.0          0.851883           7.782222   
min              8.0          0.000000           0.000000   
25%              8.0          0.000000           6.000000   
50%              8.0          1.000000          10.000000   
75%              8.0          1.000000          15.000000   
max              8.0          3.000000          40.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count            4410.000000     4410.000000              4410.000000   
mean                2.799320        7.008163                 2.187755   
std                 1.288978        6.125135                 3.221699   
min                 0.000000        0.000000                 0.000000   
25%                 2.000000        3.000000                 0.000000   
50%                 3.000000        5.000000                 1.000000   
75%                 3.000000        9.000000                 3.000000   
max                 6.000000       40.000000                15.000000   

       YearsWithCurrManager  
count           4410.000000  
mean               4.123129  
std                3.567327  
min                0.000000  
25%                2.000000  
50%                3.000000  
75%                7.000000  
max               17.000000

In [38]:
headers = ['Age','DistanceFromHome','Education','EmployeeCount','EmployeeID','JobLevel','MonthlyIncome','NumCompaniesWorked','PercentSalaryHike','StandardHours','StockOptionLevel','TotalWorkingYears','TrainingTimesLastYear','YearsAtCompany','YearsSinceLastPromotion','YearsWithCurrManager']
median = []
mean = []
mode = []
var = []
std = []
kurt = []
skew = []


### Median of each column

In [45]:
for i in data_house.median():
    print(i)
    median.append(i)

36.0
7.0
3.0
1.0
2205.5
2.0
49190.0
2.0
14.0
8.0
1.0
10.0
3.0
5.0
1.0
3.0


### Mean of each column

In [39]:
for i in data_house.mean():
    print(i)
    mean.append(i)

36.923809523809524
9.19251700680272
2.912925170068027
1.0
2205.5
2.0639455782312925
65029.31292517007
2.6948303347756775
15.209523809523809
8.0
0.7938775510204081
11.279936378095888
2.7993197278911564
7.0081632653061225
2.1877551020408164
4.12312925170068


### Mode of each column

In [51]:
data_house[['Age','DistanceFromHome','Education','MonthlyIncome','NumCompaniesWorked', 'PercentSalaryHike','TotalWorkingYears', 'TrainingTimesLastYear','YearsAtCompany','YearsSinceLastPromotion', 'YearsWithCurrManager']].mode()

Age  DistanceFromHome  Education  MonthlyIncome  NumCompaniesWorked  \
0   35                 2          3          23420                 1.0   

   PercentSalaryHike  TotalWorkingYears  TrainingTimesLastYear  \
0                 11               10.0                      2   

   YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0               5                        0                     2

### Variance of each column

In [46]:
for i in data_house.var():
    print(i)
    var.append(i)

83.41719210705376
65.69143866210581
1.0484380279669137
0.0
1621042.5
1.2247601149781753
2215480270.2241244
6.244435683052258
13.389067815831256
0.0
0.7257053059372989
60.56298145049679
1.66146472197407
37.51728421919913
10.379346512930416
12.725820098962698


### Standard Deviation of each column

In [41]:
for i in data_house.std():
    std.append(i)

### Kurtosis of each column

In [42]:
for i in data_house.kurt():
    kurt.append(i)

### Skewness of each column

In [43]:
data_house.skew()
for i in data_house.skew():
    print(i)
    skew.append(i)

0.413004952697689
0.9574657463789057
-0.28948387841167106
0.0
0.0
1.0247032298420649
1.3688841631898714
1.0267666759708942
0.8205689837508005
0.0
0.9683206728523724
1.1168317963678702
0.5527476257400715
1.763328231666419
1.9829391562991432
0.8328836111366986


In [50]:
print("|Variables|Mean|Median|Mode|Variance|Standard Deviation|Skewness|Kurtosis|")
print("| --- " * 7, "|")
for k in range(len(headers)):
    print("|", headers[k],"|", mean[k], "|", median[k], "|", "|", var[k], "|", std[k], "|", skew[k], "|", kurt[k], "|")

|Variables|Mean|Median|Mode|Variance|Standard Deviation|Skewness|Kurtosis|
| --- | --- | --- | --- | --- | --- | ---  |
| Age | 36.923809523809524 | 36.0 | | 83.41719210705376 | 9.133301271011144 | 0.413004952697689 | -0.40595053984966967 |
| DistanceFromHome | 9.19251700680272 | 7.0 | | 65.69143866210581 | 8.105025518905281 | 0.9574657463789057 | -0.2270453548764575 |
| Education | 2.912925170068027 | 3.0 | | 1.0484380279669137 | 1.0239326286269588 | -0.28948387841167106 | -0.5605690113244304 |
| EmployeeCount | 1.0 | 1.0 | | 0.0 | 0.0 | 0.0 | 0.0 |
| EmployeeID | 2205.5 | 2205.5 | | 1621042.5 | 1273.201672948948 | 0.0 | -1.1999999999999997 |
| JobLevel | 2.0639455782312925 | 2.0 | | 1.2247601149781753 | 1.1066888067465828 | 1.0247032298420649 | 0.39552536469239596 |
| MonthlyIncome | 65029.31292517007 | 49190.0 | | 2215480270.2241244 | 47068.88855947339 | 1.3688841631898714 | 1.0002318550155214 |
| NumCompaniesWorked | 2.6948303347756775 | 2.0 | | 6.244435683052258 | 2.49888688880714

<h2 align='center'>Conclusion</h2>

|Factors|Mean|Median|Mode|Variance|Standard Deviation|Skewness|Kurtosis|
| --- | --- | --- | --- | --- | --- | ---  | --- |
| **Age**| 36.923809523809524 | 36.0 |35 | 83.41719210705376 | 9.133301271011144 | 0.413004952697689 | -0.40595053984966967 |
| **DistanceFromHome** | 9.19251700680272 | 7.0 |2.0 | 65.69143866210581 | 8.105025518905281 | 0.9574657463789057 | -0.2270453548764575 |
| **Education** | 2.912925170068027 | 3.0 |3.0 | 1.0484380279669137 | 1.0239326286269588 | -0.28948387841167106 | -0.5605690113244304 |
| **EmployeeCount** | 1.0 | 1.0 |- | 0.0 | 0.0 | 0.0 | 0.0 |
| **EmployeeID** | 2205.5 | 2205.5 |- | 1621042.5 | 1273.201672948948 | 0.0 | -1.1999999999999997 |
| **JobLevel** | 2.0639455782312925 | 2.0 |- | 1.2247601149781753 | 1.1066888067465828 | 1.0247032298420649 | 0.39552536469239596 |
| **MonthlyIncome** | 65029.31292517007 | 49190.0 |23420 | 2215480270.2241244 | 47068.88855947339 | 1.3688841631898714 | 1.0002318550155214 |
| **NumCompaniesWorked** | 2.6948303347756775 | 2.0 | 1.0| 6.244435683052258 | 2.498886888807146 | 1.0267666759708942 | 0.007287480878248154 |
| **PercentSalaryHike** | 15.209523809523809 | 14.0 | 11.0| 13.389067815831256 | 3.659107516298374 | 0.8205689837508005 | -0.3026383931044334 |
| **StandardHours** | 8.0 | 8.0 | -| 0.0 | 0.0 | 0.0 | 0.0 |
| **StockOptionLevel** | 0.7938775510204081 | 1.0 |- | 0.7257053059372989 | 0.851883387522787 | 0.9683206728523724 | 0.3610859039442924 |
| **TotalWorkingYears** | 11.279936378095888 | 10.0 |10.0 | 60.56298145049679 | 7.782222140911733 | 1.1168317963678702 | 0.9129359960797361 |
| **TrainingTimesLastYear** | 2.7993197278911564 | 3.0 | 2.0| 1.66146472197407 | 1.2889781697042313 | 0.5527476257400715 | 0.49114899850209515 |
| **YearsAtCompany** | 7.0081632653061225 | 5.0 |5 | 37.51728421919913 | 6.125135444967656 | 1.763328231666419 | 3.9238642054014528 |
| **YearsSinceLastPromotion** | 2.1877551020408164 | 1.0 |0 | 10.379346512930416 | 3.2216993206893805 | 1.9829391562991432 | 3.601760518317168 |
| **YearsWithCurrManager** | 4.12312925170068 | 3.0 |2.0 | 12.725820098962698 | 3.5673267440707894 | 0.8328836111366986 | 0.16794854278414206 |
